projects/angular/src/data/datagrid/datagrid.ts

```diff
@@ -274,7 +278,19 @@ export class ClrDatagrid<T = any> implements AfterContentInit, AfterViewInit, On
           /**
            * Reopen updated row or close it
            */
-          row ? this.detailService.open(row.item, row.detailButton.nativeElement) : this.detailService.close();
+          if (row) {
+            this.detailService.open(row.item, row.detailButton.nativeElement);
+          } else if (!this.hasVirtualScroller) {
```

Comment:
We need to keep the old condition where row is not available.
```suggestion
          } else if (!this.hasVirtualScroller || !row) {
```

Reply:
Corrected.